test/test_s3.py

```diff
@@ -222,8 +222,8 @@ def test_sanity_secure(self):
         self.assertIsNotNone(s3_client)
 
     def test_sanity_network_interface_names(self):
-        s3_client = s3_client_new(True, self.region, network_interface_names=("eth0", "eth1"))
-        self.assertIsNotNone(s3_client)
+        with self.assertRaises(Exception):
+            s3_client_new(True, self.region, network_interface_names=("eth0", "invalid-network-interface"))
```

Comment:
should we also check the happy case?

Reply:
It's not easy to test since only a few platforms have a happy case, and even then, it's different for each platform. This test is only verifying that we pass the value to c-s3 correctly.

Reply:
We can add a comment here that the functionality was tested in C properly, and here we make sure the binding works.

Reply:
I think the term "sanity" in the test name already signifies this intent. I can add a comment if you still think we should include one here.

Reply:
Well, we have bunch of sanity in the name of the test in aws-c-*, they are meant to simply test it works in the happy case,. eg: https://github.com/awslabs/aws-c-http/blob/main/tests/CMakeLists.txt#L410

I don't think we have a fairly common agreement on "sanity" in the test name means.

It's fine to me. But just mentioning.